app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java

```diff
@@ -753,32 +758,103 @@ public void onNegative(MaterialDialog materialDialog) {
         b.show();
     }
 
+    static final Pattern ZIP_FILE_REGEX = Pattern.compile("[\\\\\\/:\\*\\?\"<>\\|\\x01-\\x1F\\x7F]", Pattern.CASE_INSENSITIVE);
+
+    static final class CompressFileDialogTextValidator extends SimpleTextWatcher implements View.OnFocusChangeListener
```

Comment:
This class should probably be in views (maybe extending TextView?).

Reply:
Agreed. Maybe a `View` composing `TextInputLayout` and `EditText` but reusable is a better approach.

Reply:
An XML layout!

Reply:
Sure. On the way.